app/callbacks/data_callbacks.py

```diff
@@ -52,12 +57,28 @@ def login_callback(n_clicks, username, password, user_headers):
     This function is triggered when the login button is clicked. It verifies the provided username and password,
     attempts to authenticate the user via the API, and updates the user credentials and headers.
     If authentication fails or credentials are missing, it provides appropriate feedback.
+    After login succeeds and while data required to boot the dashboard is being fetched from the API,
+    the login form is hidden and a spinner is displayed.
 
     Returns:
-        dash.dependencies.Output: Updated user credentials and headers, and form feedback.
+        dash.dependencies.Output: Updated user credentials and headers, and form feedback + styles to hide/show login elements and loading spinners.
     """
+    input_style_unchanged = {"width": "250px"}
+    empty_style_unchanged = {"": ""}
```

Comment:
Would be cleaner to user `{}` instead, but then `mypy` complains because it wants to know the type of `empty_style_unchanged` and it seems python does not infer types for empty dictionaries => using `{"": ""}` seems simpler than bringing in a whole typing library to handle that error only (`from typing import Dict, Optional ... empty_style: Dict[str, Optional[str]] = {}`).